lnwire/commit_sig.go

```diff
@@ -45,6 +45,10 @@ type CommitSig struct {
 	// being signed for. In this case, the above Sig type MUST be blank.
 	PartialSig OptPartialSigWithNonceTLV
 
+	// CustomRecords maps TLV types to byte slices, storing arbitrary data
+	// intended for inclusion in the ExtraData field.
+	CustomRecords CustomRecords
```

Comment:
We could add that to `CustomRecords.Validate()`. But I'm not sure we should do that, because for the taproot assets case, when we have many in-flight HTLCs, a custom message might be larger than 65k.

But is there an overall message limit that's centrally enforced?
I noticed that any message that has a `ExtraOpaqueData` field just reads that from the remaining bytes in the message stream, and the size isn't checked in `lnwire.ReadElement`. Do we check the full message length somewhere before we start decoding it?

Reply:
The packet length in the `brontide` code is a uint16. So if a message larger than this is sent, it'll get truncated and potentially not make sense to the receiver

Reply:
Ah, okay. Then this should be safe. The custom records are using the p2p variant of the TLV decoder, so an invalid size shouldn't lead to an over-allocation.

Reply:
Sorry, I don't think I was clear. My concern here is that we may set the custom records when the commitment transaction has many HTLCs. Then when we attempt to write the message, it won't work due to the packet size limitation.